module/VuFind/src/VuFind/View/Helper/Root/Icon.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Make link view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\Config\Config;
+use Laminas\View\Helper\AbstractHelper;
+use VuFind\Config\PluginManager as ConfigManager;
+
+/**
+ * Make link view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Icon extends AbstractHelper
+{
+    /**
+     * Default icon template
+     *
+     * @var string
+     */
+    protected $defaultSet;
+
+    /**
+     * Transforming map
+     *
+     * @var array
+     */
+    protected $iconMap;
+
+    /**
+     * Constructor
+     *
+     * @param Config $config Icons config (icons.ini)
+     */
+    public function __construct(\VuFindTheme\ThemeInfo $themeInfo)
+    {
+        $themeConfig = $themeInfo->getThemeInfo()[$themeInfo->getTheme()];
```

Comment:
I think we're going to need to make this more robust (with proper inheritance/merging), or else it's only going to work when the full configuration exists in the active theme (and not a parent of it). This should be fine for prototyping purposes, but it's worth adding a TODO checkbox about.

Reply:
Does `getThemeInfo()` not handle inheritance? It seems I misread my test, I thought inheritance was being handled. Is there no mechanism to get a fully inherited theme config of the active theme?

Reply:
`getThemeInfo` retrieves information about all of the themes in the inheritance chain, but it doesn't automatically merge them together. When you call `$themeInfo->getThemeInfo()[$themeInfo->getTheme()]`, that is going to give you the contents of theme.config.php from the current active theme... but if the data you're looking for is not in there, you'll need the use the 'extends' and 'mixins' values from that array to look up the next parent/mixin in line, and so on.

It might make sense to add a utility method to ThemeInfo like getMergedConfig() to make it easier to fetch this kind of data in one call. I can take a stab at it if that would be helpful.

Reply:
That would be helpful! I think there's some code in the theme compiler or PHP less compiler that may be helpful.

Reply:
I'll see what I can do!

Reply:
See 74c09bf -- I think this does the trick!

Reply:
Now I just need to write a unit test so our coverage in VuFindTheme doesn't drop. I've added a TODO checkbox and will take care of it closer to merge time.